### ISSUE TITLE
NavigationCamera: Following state not centering correctly

### ISSUE THREAD

**Issue by Christophe668** [mapbox/mapbox-navigation-android#5057]
**Android API:** 30
**Mapbox Navigation SDK version:** 2.0.0 (GA)

### Steps to trigger behavior
Sometimes the camera navigation goes crazy in "FOLLOWING" state.
Not sure what triggers that or how I can reproduce it yet, but happy to provide anything that could help you debug this.
From what I see on the debugger tool, new points are added for some reasons.

Not sure if this is because if this might be related to [this issue](https://github.com/mapbox/mapbox-navigation-android/issues/5051) or maybe also to alternative routes?
(By the way, I don't see options to disable alternative routes options, only `intervalMillis()` is available on `RouteAlternativesOptions`

Sorry for this unclear report. I'll document more once I know more but maybe you can help me give you the right data that you might need.

| Case 1  | Case 2 |
| ------------- | ------------- |
| <img width="388" alt="Screenshot 2021-10-28 at 13 47 43" src="https://user-images.githubusercontent.com/1552554/139251488-bee83212-1cf9-4227-8c5a-e824621a0c28.png"> | <img width="400" alt="Screenshot 2021-10-28 at 13 47 26" src="https://user-images.githubusercontent.com/1552554/139251494-75e0f40d-5e7e-41a9-bbb4-8e79a61d521f.png"> |

**Comment by LukasPaczos** [comment]
Hey @Christophe668, thanks for reaching out and reporting this. A couple of questions that might help us get closer:
- does this happen with an initial route, or only after re-routes?
- are you using `MapboxNavigationViewportDataSource#additionalPointsToFrameForFollowing`?
- Would you be able to record and share a video that showcases what happens on screen?

**Comment by Christophe668** [comment]
Thanks Lukas for answering! 

* Could not tell at the moment, will try to reproduce when I can (need to be tested in real live situation)
* I'm currently not using that method, only `additionalPointsToFrameForOverview`
* Sure, will do that if I detect this in the future

**Comment by LukasPaczos** [comment]
> Additionally, would be nice to have a method to clear the the additional points

This was envisioned to be done by passing an empty list to `additionalPointsToFrameForOverview` instead of the SDK exposing `addAdditionalPointsToFrameForOverview` and `removeAdditionalPointsToFrameForOverview` functions.

**Comment by Christophe668** [comment]
Oh ok, make sense. I think my mind reads "add" and not "additional", reflexes I guess!
